tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -111,12 +116,16 @@ func NewRKEMachinePool(controlPlaneRole, etcdRole, workerRole bool, poolName str
 }
 
 type NodeRoles struct {
```

Comment:
this stuct is getting too dense and no longer matches its name, if you use a separate solution to the above, please rename this to match what the new struct represents.